### ISSUE TITLE
hostname sanity check

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#140]
**Reported by naughtont on 23 Sep 2005 16:10 UTC**
```
There are big configuration problems if the value of
hostname doesn't match the IP address associated with
the value passed to 'install_cluster'.  I think we
should add a very basic heuristic to check
that the value of hostname matches the value of the
ethX's IP address.

Example, see attachedk perl script for a basic idea.
```

**Comment by dikim33** [comment]
**Comment by muglerj on 23 Sep 2005 17:23 UTC**
```

Looks like the most common problem is people assigning the
same name to both interfaces. Not sure why they do that, but
thats seems to be what happens.
```

**Comment by dikim33** [comment]
**Comment by bli on 31 Oct 2005 03:44 UTC**
```

Can we fast-track this bugfix?  I'm having issues with the
SGE package as well.

We can probably add the sanity check to oscar_wizard -
asking the user to make sure that the correct entry of the
host's ip is in /etc/hosts - but what if the ip is assigned
via DHCP?  Or do we just not support it.
```

**Comment by dikim33** [comment]
**Comment by naughtont on 31 Oct 2005 15:49 UTC**
```

I'm swamped right now but the code I attached should get
most of the trick, probably needs to be beat on a bit to
make sure there are no stupid corner cases.  

I believe it would go in 'install_cluster' and check/print
warning there.  I'll try to look at this later this week
when I get to oscar-head (still off living in branches and
branches of branches :( ) 
```

**Comment by dikim33** [comment]
**Comment by bli on 25 Jul 2006 01:54 UTC**
Assigning to Geoffroy since he is now responsible for the system sanity check mechanism (which has a check for hostname).

The patch mentioned can be found here:

http://sourceforge.net/tracker/index.php?func=detail&aid=1301468&group_id=9368&atid=109368

**Comment by dikim33** [comment]
**Comment by valleegr on 20 Aug 2006 21:36 UTC**
(In [5307]) This check-in fixes #142, we check now much more stuff regarding the hostname
and network configuration. It should check everything described in the bug 
report. If not please reopen the bug with the details of what you want to be
done.
